### PR TITLE
No name

### PR DESCRIPTION
**Pull request by Nekotekina**
Some refactoring.

### CODE REVIEW

**Comment by Nekotekina** [comment]
Compilers being garbage, can't accept auto& references in template parameters.
Added partial Intel JIT event support (Windows, Linux) in LLVM, untested.
Currently working on better Linux Perf JIT annotations for all JIT stuff including ASMJIT.

File: rpcs3/Emu/RSX/rsx_methods.cpp [hunk 1]
```diff
@@ -895,10 +895,9 @@ namespace rsx
 
 	namespace nv308a
 	{
-		template<u32 index>
 		struct color
 		{
-			static void impl(thread* rsx, u32 /*reg*/, u32 /*arg*/)
+			static void core(thread* rsx, u32 index, u32 x700minus_idx)
```
Comment (elad335) [inline rpcs3/Emu/RSX/rsx_methods.cpp:900]:
```suggestion
			static void core(thread* rsx, u32 rsx, u32 /*arg*/)
```
Simply use tge reg argument, index = reg - NV308A_COLOR.

Comment (Nekotekina) [reply rpcs3/Emu/RSX/rsx_methods.cpp]: Okay, that should work too.

**Comment by Nekotekina** [comment]
Improved profiling with perf a bit
image
image

**Comment by coolllman** [comment]
[RPCS3.log](https://github.com/RPCS3/rpcs3/files/7777182/RPCS3.log)
All games not boot, rpcs3 closes when game load

**Pull request merged**